server/src/server.ts

```diff
@@ -68,8 +68,10 @@ connection.onHover((_params) => {
     if (!entityName) {
         return null;
     }
-    const entity = completionProvider.getItemByLabel(entityName);
-    if (!entity || !entity.documentation) {
+    // Multiple elements can have the same shortName
+    const entities = completionProvider.getAllItemsByShortName(entityName);
+    const entity = entities.find((elem) => !!elem.documentation);
```

Comment:
What does this do? Get the first guy with nonempty doc? :thinking:

Reply:
Yes, exactly.

Reply:
### 1

Why call a “get all” method (I saw that you have equivalents with a single returned item) to eventually cut down the result to a single guy? You even lose processing power and time by scanning dudes that won’t serve any purpose. Consider, for example, giving the `(elem) => !!elem.documentation` lambda filter as a new parameter for `getAllItemsByShortName(…)` (which would become something like `getFirstItemByShortNameMatching(…)`) and stop the search as soon as you find a suitable guy.

### 2

Is it OK to pick the first? No other criterion available? :thinking:

Reply:
= 1

Good idea with the lambda thingy. I'll do this.

= 2

No, no other criterion available yet that I may think of.